iceoryx_examples/request_response/server_cxx_basic.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! [iceoryx includes]
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+//! [iceoryx includes]
+
+#include <iostream>
+
+int main()
+{
+    //! [initialize runtime]
+    constexpr char APP_NAME[] = "iox-cpp-request-response-server-basic";
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+    //! [initialize runtime]
+
+    //! [create server]
+    iox::popo::Server<AddRequest, AddResponse> server({"Example", "Request-Response", "Add"});
+    //! [create server]
+
+    //! [process requests in a loop]
+    while (!iox::posix::hasTerminationRequested())
+    {
+        //! [take request]
+        server.take().and_then([&](const auto& request) {
```

Comment:
do you want to take with a while loop? A server that assumes that one take every 100 ms will do feels a bit broken